cmd/start.go

```diff
@@ -737,13 +698,80 @@ func (x *Start) Execute(args []string) error {
 	return nil
 }
 
+func getConfigFiles(datafolder string) error {
+	var err error
+	mainConfig, err = ioutil.ReadFile(path.Join(datafolder, "config"))
+	if err != nil {
+		log.Error("read config:", err)
+		return err
+	}
+
+	apiConfig, err = schema.GetAPIConfig(mainConfig)
+	if err != nil {
+		log.Error("scan api config:", err)
+		return err
+	}
+	torConfig, err = schema.GetTorConfig(mainConfig)
+	if err != nil {
+		log.Error("scan tor config:", err)
+		return err
+	}
+	dataSharingConfig, err = schema.GetDataSharing(mainConfig)
+	if err != nil {
+		log.Error("scan data sharing config:", err)
+		return err
+	}
+	dropboxToken, err = schema.GetDropboxApiToken(mainConfig)
+	if err != nil {
+		log.Error("scan dropbox api token:", err)
+		return err
+	}
+	resolverConfig, err = schema.GetResolverConfig(mainConfig)
+	if err != nil {
+		log.Error("scan resolver config:", err)
+		return err
+	}
+	republishInterval, err = schema.GetRepublishInterval(mainConfig)
+	if err != nil {
+		log.Error("scan republish interval config:", err)
+		return err
+	}
+	walletsConfig, err = schema.GetWalletsConfig(mainConfig)
+	if err != nil {
+		log.Error("scan wallets config:", err)
+		return err
+	}
+	return nil
+}
+
+func ensureDatabaseAccess(db *db.SQLiteDatastore, datafolder string, testnet bool, coinType wi.CoinType) error {
+	if db.Config().IsEncrypted() {
+		db.Close()
+		fmt.Print("Database is encrypted, enter your password: ")
+		// nolint:unconvert
+		bytePassword, _ := terminal.ReadPassword(int(syscall.Stdin))
+		fmt.Println("")
+		pw := string(bytePassword)
+		db, err := InitializeRepo(datafolder, pw, "", testnet, time.Now(), coinType)
+		if err != nil && err != repo.ErrRepoExists {
+			return err
+		}
+		if db.Config().IsEncrypted() {
+			log.Error("Invalid password")
+			os.Exit(3)
+		}
+	}
+	return nil
+}
```

Comment:
It would be good for `ensureDatabaseAccess` checked when the db is not encrypted as well. Otherwise, this is just `ensureEncryptedDatabaseAccess`.